spec/rubocop/cop/style/empty_block_parameter_spec.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+describe RuboCop::Cop::Style::EmptyBlockParameter do
+  subject(:cop) { described_class.new(config) }
+
+  let(:config) { RuboCop::Config.new }
+
+  it 'registers an offense for an empty block parameter with do-end wtyle' do
+    expect_offense(<<-RUBY.strip_indent)
+      a do ||
+           ^^ Omit delimiters for the empty block parameters.
+      end
+    RUBY
+  end
+
+  it 'registers an offense for an empty block parameter with {} style' do
+    expect_offense(<<-RUBY.strip_indent)
+      a { || do_something }
+          ^^ Omit delimiters for the empty block parameters.
+    RUBY
+  end
+
+  it 'registers an offense for an empty block parameter with a lambda' do
+    expect_offense(<<-RUBY.strip_indent)
+      -> () { do_something }
```

Comment:
I would much rather have this as a separate cop.

Reply:
👍 I'll separate the cop into `Style/EmptyBlockParameter` and `Style/EmptyLambdaParameter`.